From 523c096fad9ebbfae2f14b122958708fa8d22326 Mon Sep 17 00:00:00 2001
From: Sinduran Sivarajan <sinduran@google.com>
Date: Mon, 3 Feb 2025 07:08:13 -0800
Subject: [PATCH 12/12] Disable "Developer options" by default for managed
 profiles.

DISABLE_TOPIC_PROTECTOR

Bug: 382064697
Test: go/work-profile-creation-developer-access
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:d5783311183adc56c87ce7f7a3ab7f561e03ecc0)
Merged-In: Ibe6b721f2552d9e72aba0582a2eed4ba87178c7c
Change-Id: Ibe6b721f2552d9e72aba0582a2eed4ba87178c7c
---
 .../android/server/pm/UserRestrictionsUtils.java   |  3 ++-
 .../devicepolicy/DevicePolicyManagerService.java   | 14 +++++++++++++-
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/pm/UserRestrictionsUtils.java b/services/core/java/com/android/server/pm/UserRestrictionsUtils.java
index 3be51c58798e..6bbd7eec7249 100644
--- a/services/core/java/com/android/server/pm/UserRestrictionsUtils.java
+++ b/services/core/java/com/android/server/pm/UserRestrictionsUtils.java
@@ -212,7 +212,8 @@ public class UserRestrictionsUtils {
      * in settings. So it is handled separately.
      */
     private static final Set<String> DEFAULT_ENABLED_FOR_MANAGED_PROFILES = Sets.newArraySet(
-            UserManager.DISALLOW_BLUETOOTH_SHARING
+            UserManager.DISALLOW_BLUETOOTH_SHARING,
+            UserManager.DISALLOW_DEBUGGING_FEATURES
     );
 
     /**
diff --git a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
index 56b2cb8f3a0b..3c9db643ce68 100644
--- a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
+++ b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
@@ -8124,7 +8124,8 @@ public class DevicePolicyManagerService extends BaseIDevicePolicyManager {
                 throw new IllegalArgumentException("Not active admin: " + who);
             }
 
-            if (isAdb()) {
+            boolean isAdb = isAdb();
+            if (isAdb) {
                 // Log profile owner provisioning was started using adb.
                 MetricsLogger.action(mContext, PROVISIONING_ENTRY_POINT_ADB, LOG_TAG_PROFILE_OWNER);
                 DevicePolicyEventLogger
@@ -8148,6 +8149,17 @@ public class DevicePolicyManagerService extends BaseIDevicePolicyManager {
                             UserRestrictionsUtils.getDefaultEnabledForManagedProfiles());
                     ensureUnknownSourcesRestrictionForProfileOwnerLocked(userHandle, admin,
                             true /* newOwner */);
+                    if (isAdb) {
+                        // DISALLOW_DEBUGGING_FEATURES is being added to newly-created
+                        // work profile by default due to b/382064697 . This would have
+                        //  impacted certain CTS test flows when they interact with the
+                        // work profile via ADB (for example installing an app into the
+                        // work profile). Remove DISALLOW_DEBUGGING_FEATURES here to
+                        // reduce the potential impact.
+                        admin.ensureUserRestrictions().putBoolean(
+                            UserManager.DISALLOW_DEBUGGING_FEATURES, false);
+                        saveUserRestrictionsLocked(userHandle);
+                    }
                 }
                 sendOwnerChangedBroadcast(DevicePolicyManager.ACTION_PROFILE_OWNER_CHANGED,
                         userHandle);
-- 
2.40.1

